modules/mbedtls/Kconfig.psa.logic

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2024 BayLibre SAS
+# SPDX-License-Identifier: Apache-2.0
+
+# This file extends Kconfig.psa (which is automatically generated) by adding
+# some logic between PSA_WANT symbols.
+
+# Start by defining some internal symbols that can be used to check whether there
+# is some kind of support in the PSA Crypto API.
+
+config PSA_CAN_SOME_HASH
+	bool
+	default y if (PSA_WANT_ALG_MD5      || \
+		      PSA_WANT_ALG_SHA_1    || \
+		      PSA_WANT_ALG_SHA_224  || \
+		      PSA_WANT_ALG_SHA_256  || \
+		      PSA_WANT_ALG_SHA_384  || \
+		      PSA_WANT_ALG_SHA_512  || \
+		      PSA_WANT_ALG_SHA3_224 || \
+		      PSA_WANT_ALG_SHA3_256 || \
+		      PSA_WANT_ALG_SHA3_384 || \
+		      PSA_WANT_ALG_SHA3_512)
+	help
+	  Promptless symbol that can be used to check if there's any type of hash
+	  support enabled (no matter which algorithm) in the PSA Crypto API.
+
+config PSA_CAN_SOME_ECC
+	bool
+	default y if (PSA_WANT_ECC_BRAINPOOL_P_R1_256 || \
+		      PSA_WANT_ECC_BRAINPOOL_P_R1_384 || \
+		      PSA_WANT_ECC_BRAINPOOL_P_R1_512 || \
+		      PSA_WANT_ECC_MONTGOMERY_255 || \
+		      PSA_WANT_ECC_MONTGOMERY_448 || \
+		      PSA_WANT_ECC_SECP_K1_192 || \
+		      PSA_WANT_ECC_SECP_K1_256 || \
+		      PSA_WANT_ECC_SECP_R1_192 || \
+		      PSA_WANT_ECC_SECP_R1_224 || \
+		      PSA_WANT_ECC_SECP_R1_256 || \
+		      PSA_WANT_ECC_SECP_R1_384 || \
+		      PSA_WANT_ECC_SECP_R1_521)
+	help
+	  Promptless symbol that can be used to check if there's at least one family
+	  of elliptic curve enabled in the PSA Crypto API.
+
+config PSA_CAN_ECDSA
+	bool
+	default y if (PSA_WANT_ALG_ECDSA || PSA_WANT_ALG_DETERMINISTIC_ECDSA) && \
+		      PSA_CAN_SOME_ECC
+	help
+	  Promptless symbol that can be used to check if the PSA Crypto API support
+	  ECDSA (either deterministic or not) for at least one curve family.
+
+config PSA_CAN_ECDH
+	bool
+	default y if PSA_WANT_ALG_ECDH && PSA_CAN_SOME_ECC
+	help
+	  Promptless symbol that can be used to check if the PSA Crypto API support
+	  ECDH (either deterministic or not) for at least one curve family.
+
+config PSA_CAN_ECJPAKE
+	bool
+	default y if PSA_WANT_ECC_SECP_R1_256 && PSA_WANT_ALG_SHA_256
+	help
+	  Promptless symbol that can be used to check if the PSA Crypto API support
+	  EC-JPAKE.
+
+# Dependencies between KDF (key derivation function) algorithms and low-level
+# crypto ones.
```

Comment:
If you are making such a comment, then it would make sense to properly cover all the sections. Because I think that this section stops at `PSA_WANT_ALG_TLS12_PRF`?
(Or just keep the old comment, IMO it was good, with a reference to the PSA Crypto spec.)